install_files/securedrop-app-code/DEBIAN/postinst

```diff
@@ -16,8 +17,55 @@ set -x
 # for details, see http://www.debian.org/doc/debian-policy/ or
 # the debian-policy package
 
+database_migration() {
+    database_dir='/var/lib/securedrop'
+    database_file="$database_dir/db.sqlite"
+
+    # Ensure the database exists.
+    if [ ! -e "$database_file" ]; then
+        old_umask=$(umask)
+        umask 0027 # to get os 0640 permissions from the sqlite command
+
+        # Future-proof way to create an empty database.
+        sqlite3 "$database_file" '.databases'
+        umask "$old_umask"
+        chown www-data:www-data "$database_file"
+    fi
+
+    cd '/var/www/securedrop/'
+
+    # `alembic current` outputs nothing if the DB isn't under verion control.
+    if [ -z "$(alembic current)" ]; then
+        sqlite_tables=$(sqlite3 "$database_file" .tables)
+
+        if ! echo "$sqlite_tables" | grep -q journalists; then
+            # This is a fresh install, run the full migrations.
+            alembic upgrade head
+        else
+            # This is an old install, so mark that the schema is up to date
+            # as of the version of migration 15ac9509fc68_init.py when we first
+            # started version controlling the database.
+            alembic stamp 15ac9509fc68
+        fi
+    fi
+
+    if ! alembic current | grep -q '(head)'; then
+        date=$(date +%Y-%m-%d-%H-%M-%S)
+        cp /var/lib/securedrop/db.sqlite "/var/lib/securedrop/backups/$date-db.sqlite"
```

Comment:
Some of these databases might not be _that_ small, so especially for long-running instances, we'll need some way to clean up the database backups. How about: once the database migration occurs successfully, we delete the backup? (in `postinst`)